kimchi_base/src/motor_driver.cpp

```diff
@@ -0,0 +1,127 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#include "kimchi_base/motor_driver.h"
+
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+
+namespace kimchi_base {
+
+void MotorDriver::Setup(const std::string& serial_device, int32_t baud_rate, int32_t timeout_ms) {
+  timeout_ms_ = timeout_ms;
+  try {
+    serial_port_.Open(serial_device);
+  } catch (std::exception& e) {
+    std::cout << e.what() << std::endl;
+  }
+  // TODO: Use baud_rate from parameter.
+  if (baud_rate != 57600) {
+    std::cerr << "A baudrate different than 57600 is not supported yet." << std::endl;
+  }
+  // Configure the serial port.
+  serial_port_.SetBaudRate(LibSerial::BaudRate::BAUD_57600);
+  serial_port_.SetCharacterSize(LibSerial::CharacterSize::CHAR_SIZE_8);
+  serial_port_.SetParity(LibSerial::Parity::PARITY_NONE);
+  serial_port_.SetStopBits(LibSerial::StopBits::STOP_BITS_1);
+  serial_port_.SetFlowControl(LibSerial::FlowControl::FLOW_CONTROL_NONE);
+  serial_port_.SetDTR(false);
+  serial_port_.SetRTS(false);
+  // Flush buffers.
+  serial_port_.FlushIOBuffers();
+}
+
+bool MotorDriver::is_connected() const { return serial_port_.IsOpen(); }
+
+void MotorDriver::SendEmptyMsg() { std::string response = SendMsg(""); }
+
+std::optional<MotorDriver::Encoders> MotorDriver::ReadEncoderValues() {
+  std::optional<Encoders> output;
+
+  static const std::string delimiter = " ";
+
+  const std::string response = SendMsg("e");
+
+  const size_t del_pos = response.find(delimiter);
+  const std::string token_1 = response.substr(0, del_pos).c_str();
+  const std::string token_2 = response.substr(del_pos + delimiter.length()).c_str();
+
+  size_t pos1, pos2;
+  int left_encoder_value = std::stoi(token_1, &pos1);
+  if(pos1 != token_1.length()) {
+    std::cerr << "Error parsing token_1: " << token_1 << std::endl;
+    return std::nullopt;
+  } 
+
+  int right_encoder_value = std::stoi(token_2, &pos2);
+  // -2 because of /r/n
+  if(pos2 != token_2.length() - 2) {
+    std::cerr << "Error parsing token_2: " << token_2 << std::endl;
+    std::cerr << "pos2: " << pos2 << "token_2.length(): " << token_2.length() << std::endl;
+    return std::nullopt;
+  } 
+
+  output = Encoders{left_encoder_value, right_encoder_value};
+  return output;
+}
+
+
```

Comment:
please run a formater in all cpp and header files. 

You can do it with this command:

```
clang-format -style=google -i -fallback-style=none <files>

```